### PR TITLE
x-pack/filebeat/input/httpjson: Apply rate limiting to all responses

### LINKED ISSUE #36207: x-pack/filebeat/input/httpjson/rate_limiter.go: Rate limit evaluation only happens on 429 responses
The rate limiter only evaluates the rate limit logic if a 429 response is found:

```go
func (r *rateLimiter) execute(ctx context.Context, f func() (*http.Response, error)) (*http.Response, error) {
	for {
		resp, err := f()
		if err != nil {
			return nil, fmt.Errorf("failed to read http.response.body: %w", err)
		}

		if r == nil || resp.StatusCode == http.StatusOK {
			return resp, nil
		}

		if resp.StatusCode != http.StatusTooManyRequests {
			return nil, fmt.Errorf("http request was unsuccessful with a status code %d", resp.StatusCode)
		}

		if err := r.applyRateLimit(ctx, resp); err != nil {
			return nil, err
		}
	}
}
```

this introduces several issues:

1- Some APIs do not honor the 429 convention and in that case is impossible to trigger a rate limiting event
2- the `early_limit` option is never evaluated prior to the 429 status, making it useless for its initial intention

### PR DESCRIPTION
**Pull request by chrisberkhout**
## Draft discussion

I didn't find a solution to the failing `request_honors_rate_limit` test. The context is okay when processing the first response, and remains okay after the wait time is applied, then it appears as cancelled after the retry request is run (that is, after [this line](https://github.com/chrisberkhout/elastic-beats/blob/23b974983b113318371dff52222cbc326ff69c69/x-pack/filebeat/input/httpjson/rate_limiter.go#L43)).

## Proposed commit message

```
x-pack/filebeat/input/httpjson: Apply rate limiting to all responses (#)

- Drain the body before rate limiting.
- Apply rate limiting to all responses, waiting immediately. Retry if
  the response was not successful and there was a rate limit wait (even
  if immediately expired), otherwise return.
- Improve names of variables `epoch` and `activeLimit`.
```

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Related issues

- Closes #36207

### CODE REVIEW

**Comment by ShourieG** [comment]
@chrisberkhout Can we get a Changelog update, otherwise looks pretty good to me, I'm approving to unblock, please add the Changelog before merging.

**Review by ShourieG — APPROVED** [review]
Approving to unblock, LGTM

**Comment by efd6** [comment]
Missed the changelog. Will add in new PR.

**Pull request merged**
